tests/expressions/models.py

```diff
@@ -6,8 +6,14 @@
 from django.db import models
 
 
+class Secretary(models.Model):
+    name = models.CharField(max_length=50)
+    based_in_eu = models.BooleanField(default=False)
+
+
 class Manager(models.Model):
     name = models.CharField(max_length=50)
+    secretary = models.ForeignKey(Secretary, models.CASCADE, null=True)
```

Comment:
I just realised we could do the same test by having this point to Employee, then we don't need the Secretary model.

Reply:
well, Changing `Secretary` to `Employee` will lead to `used-before-assignment` error. Putting `Manager` class definition under `Employee` class definition will not fix it and vice versa...

Reply:
I don't know what a `used-before-assignment` error is but you can definitely point secretary to `Employee`. The problem then becomes choosing something to compare through the outer ref chain. You could simply use pks and then confirm the query doesn't result in an exception. Or we could just add the based in EU flag on Employee.

Reply:
You can use 
```suggestion
    secretary = models.ForeignKey("Employee", models.CASCADE, null=True, related_name="secretaries")
```

Reply:
lgtm, will give a shot